TypeCobol/Compiler/Scanner/Token.cs

```diff
@@ -445,14 +445,50 @@ public bool CompareForReplace(Token? comparisonToken)
 
                 var endIndex = startIndexFound + comparisonToken.NormalizedText.Length - 1;
 
-                //Check if comparisonToken.NormalizedText begin/end with replace separator or is surrounded with replace separator 
-                return (startIndexFound == 0 || CobolChar.IsReplaceSeparator(NormalizedText[startIndexFound - 1]) || CobolChar.IsReplaceSeparator(comparisonToken.NormalizedText[0]))
-                       && (endIndex >= NormalizedText.Length - 1 || CobolChar.IsReplaceSeparator(NormalizedText[endIndex + 1]) || CobolChar.IsReplaceSeparator(comparisonToken.NormalizedText[comparisonToken.NormalizedText.Length-1]));
+                if (leading)
+                {
+                    //Check only char before
+                    return startIndexFound == 0 || CobolChar.IsReplaceSeparator(NormalizedText[startIndexFound - 1]);
+                } 
+                else if (trailing)
+                {
+                    return endIndex >= NormalizedText.Length - 1 || CobolChar.IsReplaceSeparator(NormalizedText[endIndex + 1]);
+                }
+                else
+                {
+                    //Check if comparisonToken.NormalizedText begin/end with replace separator or is surrounded with replace separator 
+                    return (startIndexFound == 0 || CobolChar.IsReplaceSeparator(NormalizedText[startIndexFound - 1]) || CobolChar.IsReplaceSeparator(comparisonToken.NormalizedText[0]))
+                           && (endIndex >= NormalizedText.Length - 1 || CobolChar.IsReplaceSeparator(NormalizedText[endIndex + 1]) || CobolChar.IsReplaceSeparator(comparisonToken.NormalizedText[comparisonToken.NormalizedText.Length - 1]));
+                }
             }
 
             //Text-based comparison for AlphanumericLiteral, NumericLiteral, Symbol and SyntaxLiteral families
             if (this.TokenFamily == comparisonToken.TokenFamily && IsFamilyComparable())
             {
+                var startIndexFound = NormalizedText.IndexOf(comparisonToken.NormalizedText, StringComparison.OrdinalIgnoreCase);
+                if (startIndexFound < 0)
+                {
+                    return false;
+                }
+
+                //PartialCobolWord are surrounded with separator for replace, no need to manually check
+                if (comparisonToken.TokenType == TokenType.PartialCobolWord)
+                {
+                    return true;
+                }
+
+                var endIndex = startIndexFound + comparisonToken.NormalizedText.Length - 1;
+
+                if (leading)
+                {
+                    //Check only char before
+                    return startIndexFound == 0 || CobolChar.IsReplaceSeparator(NormalizedText[startIndexFound - 1]);
+                }
+                else if (trailing)
+                {
+                    return endIndex >= NormalizedText.Length - 1 || CobolChar.IsReplaceSeparator(NormalizedText[endIndex + 1]);
+                }
+
```

Comment:
I need to review that in VS but it looks like it could be factorized with previous code.